lib/vrl/parser/src/lex.rs

```diff
@@ -908,7 +908,13 @@ impl<'input> Lexer<'input> {
     fn numeric_literal(&mut self, start: usize) -> SpannedResult<'input, usize> {
```

Comment:
```suggestion
    fn numeric_literal_or_identifier(&mut self, start: usize) -> SpannedResult<'input, usize> {
```

It would be nice if this function can only ever return a numeric literal (or an error), but given that this is no longer the case after this PR, let's at least update its name to reflect this (similar to `identifier_or_function_call`).